Objective-C/CBLDatabase.mm

```diff
@@ -493,6 +493,9 @@ + (BOOL) copyFromPath: (NSString*)path
     slice fromPath(path.fileSystemRepresentation);
     CBLStringBytes destinationName(name);
     C4DatabaseConfig2 c4Config = c4DatabaseConfig2(config ?: [CBLDatabaseConfiguration new]);
+    CBLStringBytes d(config.directory);
+    c4Config.parentDirectory = d;
+    
```

Comment:
I'm confused with the behavior (I feel that maybe CBLStringBytes is the source of the issue?).  If later trying to get c4Config from the c4database, will the parentDirectory still **always** be valid?

Reply:
I have just looked at CBLStringBytes class and the code here. I think the problem happens when the string is stored in the _local array (the other case shouldn't cause the issue as the live of the string's data is bound to the live of the config.directory string.  The problem here is that the struct will be copied when the struct is returned. The original struct will be deleted and the _local array will be freed as well. CBLStringBytes would need a copy constructor that copies the _local array.

Reply:
I can think of one scenario, where static function does reserve the c4Config.parentDirectory string when its first used. Later on when second c4Config with different directory is initialized it will try to use the first directory string.

Reply:
I think the real issue is in `CBLStringBytes` when copying the CBLStringBytes that has the raw string value in _local[]. Implementing a copy constructor/assignment in `CBLStringBytes` to properly _local[] would fix the issue here.

Reply:
ok. let me try it.

Reply:
I was wrong about suggesting to provide a copy constructor to copy the _local array. In general, the array should be copied by default. The real problem here that prevent _local to be copied because we have assign the CBLStringBytes to a slice variable which will cause the slice() operation to be used hence no the copying of CBLStringBytes.

I couldn't think a good way around this and I'm wondering how much performance would we get out of the getBytes() logic as I expect that CFStringGetCStringPtr would normally works.